test/e2e/gateway/gateway_test.go

```diff
@@ -99,20 +107,32 @@ var _ = Describe("Gateway API", func() {
 
 		// Wait for Envoy to be healthy.
 		require.NoError(f.T(), f.Deployment.WaitForEnvoyDaemonSetUpdated())
+
+		f.CreateGatewayClassAndWaitFor(contourGatewayClass, gatewayClassValid)
+		f.CreateGatewayAndWaitFor(contourGateway, gatewayValid)
 	})
 
 	AfterEach(func() {
+		require.NoError(f.T(), f.DeleteGatewayClass(contourGatewayClass, false))
 		require.NoError(f.T(), f.Deployment.StopLocalContour(contourCmd, contourConfigFile))
 	})
 
 	Describe("HTTPRoute: Insecure (Non-TLS) Gateway", func() {
 		testWithHTTPGateway := func(body e2e.NamespacedTestBody) e2e.NamespacedTestBody {
+			gatewayClass := &gatewayv1alpha1.GatewayClass{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: fmt.Sprintf("contour-class-%d", getRandomNumber()),
+				},
+				Spec: gatewayv1alpha1.GatewayClassSpec{
+					Controller: fmt.Sprintf("projectcontour.io/ingress-controller-%d", getRandomNumber()),
+				},
+			}
```

Comment:
looks like the `GatewayClass` is always the same, you could extract a helper to get an instance of it